### PR TITLE
`azurerm_batch_pool` - fixes `TestAccBatchPool_extensions`

### PR DESCRIPTION
**Pull request by liuwuliuyun**
`azurerm_batch_pool`
- fixes `TestAccBatchPool_extensions`
- fixes https://github.com/hashicorp/terraform-provider-azurerm/issues/20530
- change test case content to make it functional

Testing Evidence:
```
GOROOT=C:\Program Files\Go #gosetup
GOPATH=C:\Users\yunliu1\go #gosetup
"C:\Program Files\Go\bin\go.exe" test -c -o C:\Users\yunliu1\AppData\Local\JetBrains\GoLand2023.2\tmp\GoLand\___2TestAccBatchPool_extensions_in_github_com_hashicorp_terraform_provider_azurerm_internal_services_batch.test.exe github.com/hashicorp/terraform-provider-azurerm/internal/services/batch #gosetup
"C:\Program Files\Go\bin\go.exe" tool test2json -t C:\Users\yunliu1\AppData\Local\JetBrains\GoLand2023.2\tmp\GoLand\___2TestAccBatchPool_extensions_in_github_com_hashicorp_terraform_provider_azurerm_internal_services_batch.test.exe -test.v -test.paniconexit0 -test.run ^\QTestAccBatchPool_extensions\E$ #gosetup
=== RUN   TestAccBatchPool_extensions
=== PAUSE TestAccBatchPool_extensions
=== CONT  TestAccBatchPool_extensions
--- PASS: TestAccBatchPool_extensions (587.49s)
PASS


Process finished with the exit code 0
```

### CODE REVIEW

**Review by WodansSon — APPROVED** [review]
Hi @liuwuliuyun, thanks for opening this PR. I've given it a look an this LGTM. 🚀

File: internal/services/batch/batch_pool.go [hunk 1]
```diff
@@ -868,11 +869,17 @@ func expandBatchPoolExtension(ref map[string]interface{}) (*pool.VmExtension, er
 	}
 
 	if settings, ok := ref["settings_json"]; ok {
-		result.Settings = pointer.To(settings)
+		err := json.Unmarshal([]byte(settings.(string)), &result.Settings)
+		if err != nil {
+			return nil, err
```
Comment (stephybun) [inline internal/services/batch/batch_pool.go:874]: Can we wrap this error for a bit more info
```suggestion
			return nil, fmt.Errorf("unmarshaling `settings_json`: %+v", err)
```

File: internal/services/batch/batch_pool.go [hunk 2]
```diff
@@ -868,11 +869,17 @@ func expandBatchPoolExtension(ref map[string]interface{}) (*pool.VmExtension, er
...
+		}
 	}
 
 	if protectedSettings, ok := ref["protected_settings"]; ok {
-		result.ProtectedSettings = pointer.To(protectedSettings)
+		err := json.Unmarshal([]byte(protectedSettings.(string)), &result.ProtectedSettings)
+		if err != nil {
+			return nil, err
```
Comment (stephybun) [inline internal/services/batch/batch_pool.go:881]:
```suggestion
			return nil, fmt.Errorf("unmarshaling `protected_settings`: %+v", err)
```

**Review by stephybun — CHANGES_REQUESTED** [review]
Two minor comments on the error messages, would you mind fixing those? This should be good to go then.

**Review by stephybun — APPROVED** [review]
Thanks @liuwuliuyun LGTM 👍

**Pull request merged**